pkg/controller/logstash/env_test.go

```diff
@@ -0,0 +1,106 @@
+	logstashv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/logstash/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
+)
+
+func Test_getEnvVars(t *testing.T) {
+	fakeLogstashUserSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Name: "logstash-sample-default-elasticsearch-sample-logstash-user", Namespace: "default"},
+		Data:       map[string][]byte{"default-logstash-sample-default-elasticsearch-sample-logstash-user": []byte("1234567890")},
+	}
+
+	params := Params{
+		Logstash: logstashv1alpha1.Logstash{
+			Spec: logstashv1alpha1.LogstashSpec{
+				ElasticsearchRefs: []commonv1.ObjectSelector{
+					{Name: "elasticsearch-sample", Namespace: "default"},
+				},
+			},
+		},
+		Client:  k8s.NewFakeClient(&fakeLogstashUserSecret),
+		Context: context.Background(),
+	}
+
+	for _, tt := range []struct {
+		name          string
+		params        Params
+		setAssocConfs func(assocs []commonv1.Association)
+		wantEnvs      []corev1.EnvVar
+	}{
+		{
+			name: "no es ref",
+			params: Params{
+				Logstash: logstashv1alpha1.Logstash{
+					Spec: logstashv1alpha1.LogstashSpec{},
+				},
+				Client:  k8s.NewFakeClient(),
+				Context: context.Background(),
+			},
+			setAssocConfs: func(assocs []commonv1.Association) {},
+			wantEnvs:      []corev1.EnvVar(nil),
+		},
+		{
+			name:   "es ref",
+			params: params,
+			setAssocConfs: func(assocs []commonv1.Association) {
+				assocs[0].SetAssociationConf(&commonv1.AssociationConf{
+					AuthSecretName: "logstash-sample-default-elasticsearch-sample-logstash-user",
+					AuthSecretKey:  "default-logstash-sample-default-elasticsearch-sample-logstash-user",
+					CACertProvided: true,
+					CASecretName:   "logstash-sample-logstash-es-default-elasticsearch-sample-ca",
+					URL:            "https://elasticsearch-sample-es-http.default.svc:9200",
+					Version:        "8.7.0",
+				})
+				assocs[0].SetNamespace("default")
+			},
+			wantEnvs: []corev1.EnvVar{
+				{Name: "DEFAULT_ELASTICSEARCH_SAMPLE_ES_HOSTS", Value: "https://elasticsearch-sample-es-http.default.svc:9200"},
+				{Name: "DEFAULT_ELASTICSEARCH_SAMPLE_ES_USERNAME", Value: "default-logstash-sample-default-elasticsearch-sample-logstash-user"},
+				{Name: "DEFAULT_ELASTICSEARCH_SAMPLE_ES_PASSWORD", Value: "1234567890"},
+				{Name: "DEFAULT_ELASTICSEARCH_SAMPLE_ES_SSL_CERTIFICATE_AUTHORITY", Value: "/mnt/elastic-internal/elasticsearch-association/default/elasticsearch-sample/certs/ca.crt"},
+			},
+		},
+		{
+			name:   "es ref without tls",
+			params: params,
+			setAssocConfs: func(assocs []commonv1.Association) {
+				assocs[0].SetAssociationConf(&commonv1.AssociationConf{
+					AuthSecretName: "logstash-sample-default-elasticsearch-sample-logstash-user",
+					AuthSecretKey:  "default-logstash-sample-default-elasticsearch-sample-logstash-user",
+					CACertProvided: false,
+					CASecretName:   "logstash-sample-logstash-es-default-elasticsearch-sample-ca",
```

Comment:
Would we expect this to be created if we are not using tls?

Reply:
removed